### PR TITLE
docs: Improve doc for gp3 storage throughput and iops

### PR DESCRIPTION
**Pull request by etiennechabert**
## Description
Adding documentation regarding this limitation for `gp3`:
https://github.com/hashicorp/terraform-provider-aws/pull/28361

## Motivation and Context
Relates to this initial issue:
https://github.com/hashicorp/terraform-provider-aws/issues/28271#issuecomment-1352059236

An easy mistake that need to be documented, since more people will try to adopt `gp3` very soon.

## Breaking Changes
Just a documentation change

## How Has This Been Tested?
Just a documentation change

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -202,6 +202,7 @@ Users have the ability to:
 
 1. This module does not create RDS security group. Use [terraform-aws-security-group](https://github.com/terraform-aws-modules/terraform-aws-security-group) module for this.
 2. By default, the variable `create_random_password` is set to true. Therefore, even if the user provides a password, it will not be read. The `create_random_password` variable should be set to false and the `password` variable should have a non-null value to be read and used.
+3. For an RDS instance with `storage_type` using `gp3`, be aware that `iops` and `storage_throughput` cannot be specified if the `allocated_storage` value is below a per-`engine` threshold. See the [RDS User Guide](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Storage.html#gp3-storage) for details.
```
Comment (etiennechabert) [inline README.md:205]: I was not sure if it was worth a new note, but also could not put this information directly in the variable documentation.

Open to a better proposal.

Comment (bryantbiggs) [reply README.md:205]: we'll try it

**Review by bryantbiggs — APPROVED** [review]
thank you!

**Comment by antonbabenko** [comment]
This PR is included in version 5.2.2 :tada:

**Pull request merged**